test/scripts/integration-tests.sh

```diff
@@ -0,0 +1,429 @@
+
+    # Check if port is available
+    if lsof -i:$SERVER_PORT >/dev/null 2>&1; then
+        echo "❌ Port $SERVER_PORT is already in use" >&2
+        exit 1
+    fi
+
+    clojure -M:run -c "$CONFIG_PATH" >"$LOG_FILE" 2>&1 &
+    local pid=$!
+    echo "📝 Server started with PID: $pid" >&2
+
+    # Give it a moment to start up
+    echo "⏳ Waiting for moclojer server to start..." >&2
+    sleep 10
+
+    # Check if server is running
+    if ! ps -p $pid >/dev/null; then
+        echo "❌ moclojer server failed to start. Check logs:" >&2
+        cat "$LOG_FILE" >&2
+        exit 1
+    fi
+
+    # Check if server is responding
+    MAX_RETRIES=5
+    RETRY_COUNT=0
+    SERVER_READY=false
+
+    while [ $RETRY_COUNT -lt $MAX_RETRIES ] && [ "$SERVER_READY" = false ]; do
+        if curl -s -o /dev/null -w "%{http_code}" --max-time $TIMEOUT "http://$SERVER_HOST:$SERVER_PORT/hello/testuser" | grep -q "200"; then
+            SERVER_READY=true
+            echo "✅ moclojer server is up and running at http://$SERVER_HOST:$SERVER_PORT!" >&2
+        else
+            echo "⏳ Waiting for server to be ready... (attempt $((RETRY_COUNT + 1))/$MAX_RETRIES)" >&2
+            RETRY_COUNT=$((RETRY_COUNT + 1))
+            sleep 5
+        fi
+    done
+
+    if [ "$SERVER_READY" = false ]; then
+        echo "❌ moclojer server did not respond in time. Check logs:" >&2
+        cat "$LOG_FILE" >&2
+        exit 1
+    fi
+
+    # Return only the PID, nothing else
+    echo "$pid"
+}
+
+# Function to stop the server
+stop_server() {
+    local pid=$1
+    if [[ -n "$pid" && "$pid" =~ ^[0-9]+$ ]]; then
+        echo "🛑 Stopping moclojer server (PID: $pid)..."
+        kill "$pid" 2>/dev/null || true
+
+        # Wait for process to actually stop
+        local max_wait=5
+        local waited=0
+        while ps -p "$pid" >/dev/null 2>&1 && [ $waited -lt $max_wait ]; do
+            sleep 1
+            waited=$((waited + 1))
+        done
+
+        # Force kill if still running
+        if ps -p "$pid" >/dev/null 2>&1; then
+            echo "⚠️ Server did not stop gracefully, force killing..."
+            kill -9 "$pid" 2>/dev/null || true
+        fi
+    else
+        echo "⚠️ Invalid PID: '$pid'. Trying to kill by process name..."
+        pkill -f "clojure -M:run -c $CONFIG_PATH" || true
+    fi
+
+    # Print server logs for debugging
+    if [ -f "$LOG_FILE" ]; then
+        echo "📋 moclojer server logs:"
+        cat "$LOG_FILE"
+    fi
+}
+
+# Generic function to test endpoints
+test_endpoint() {
+    local endpoint="$1"
+    local method="${2:-GET}"
+    local data="$3"
+    local expected_output="$4"
+    local query_params="${5:-}"
+
+    echo "🔍 Testing $method $endpoint endpoint..."
+
+    # Special case for hello-world endpoint
+    if [ "$endpoint" = "/hello-world" ]; then
+        # Execute the request
+        local url="http://$SERVER_HOST:$SERVER_PORT$endpoint"
+        local curl_opts=(-s --max-time "$TIMEOUT" -w "\n%{http_code}")
```

Comment:
just checking my understanding, this problem would not happen if the users curl with content type json right?

Reply:
I believe this commit covers the case you shared with me

https://github.com/moclojer/moclojer/pull/312/commits/9ef848ad5ec4f6492c77aed442102067646cbf46